### ISSUE TITLE
Using Transactional Templates With The SMTP API

### ISSUE THREAD

**Issue by reagan83** [sendgrid/docs#1755]
This doc needs to be improved. It makes many references in the "Body and Subject" tags section that are not expanded on in the doc.

For example (inline comments below).

Body and Subject Tags
Enabling a transactional template means that the subject and body content of your message will behave differently.
(reagan83): Ok, how differently?

If you want only the message’s content to be displayed, populate only the token in the template’s field.
(reagan83): what token in the template's field? There is no reference to 'token' anywhere in this doc other than in this sentence. Can we get an example?

If you want only the template’s content to be displayed, leave the message field (subject or body) empty, and the template will populate.
(reagan83): I can imagine most use cases will only want the template's content displayed (that's why we're using templates and in this doc section).

Text or HTML Templates?

It is best practice to provide content for both the html and the text properties in all of your emails.
(reagan83): Wait, what? Above you just said we could leave these fields empty to make use of template content.

The text property is substituted into the <%body%> of the text template and html is substituted into the <%body%> of the HTML template. If the text property is present, but not html, then the resulting email will only contain the text version of the template, not the HTML version. The <%subject%> property is used for both Text and HTML templates.

(reagan83): there are 0 examples on this page of using the <%body%> tag.  

Thanks for taking a look at these comments. Please improve this section of the docs.

https://sendgrid.com/docs/API_Reference/Web_API_v3/Transactional_Templates/smtpapi.html

**Comment by ChristophAnastasiades** [comment]
I agree, the documentation is really poor, especially the templating part. What is a text template supposed to be in contrast to an HTML template? Is there a way to set an independent template for the text/plain content?

**Comment by thinkingserious** [comment]
Hello @reagan83,<br /><br/>Thanks for your suggestion or question!<br />                                              <br />In an attempt to clear out our backlog,                                              we are closing out issues over 90 days old.                                             If this is still an issue, please feel free to reopen it.                                             <br /><br />Team SendGrid DX-Docs

**Comment by reagan83** [comment]
Seriously? That's a pretty unfortunate way to handle a backlog. As can be seen from my original thread from the low quality of your docs, and the follow-up you've done here, Sendgrid quality appears to be on the major decline.

**Comment by mbernier** [comment]
I have reopened this issue because it is valid.

**Comment by ksigler7** [comment]
I think I am going to hijack this issue for completely redoing this page. So the to do here is to rewrite the [Using Transactional Templates With the SMTP API](https://sendgrid.com/docs/API_Reference/Web_API_v3/Transactional_Templates/smtpapi.html).

Acceptance criteria:
New title: Sending a Transactional Template Email over SMTP
Includes these sections: 
    - Before you begin: this links out to other doc about required things you need to do before you can complete the steps on this page (like creating a transactional template)
    - Sending a test email (with steps and most steps having verified working code example) - these steps should cover everything you have to do to A to Z send a transactional template over SMTP, including but not limited to setting the version, and enabling the subject and body tags correctly. 
    - Advanced options: include steps and instructions for using things like substitution and section tags

**Comment by matthieunelmes** [comment]
@ksigler7  Where would be the best place to submit a feature request for HTML templating?

**Comment by mbernier** [comment]
@matthieunelmes send me an email [my github user] @ sendgrid dot com

**Comment by duncanleung** [comment]
I'll take this one up!

**Comment by duncanleung** [comment]
@ksigler7 I wanted to clarify, where should the new, _Sending a Transactional Template Email over SMTP_, page go?

I'm assuming under the **[SMTP API](https://sendgrid.com/docs/for-developers/sending-email/)** section?
Home > For Developers > Sending Email > Sending a Transactional Template Email over SMTP

**Comment by duncanleung** [comment]
@ksigler7 I'm going to start working on this in the meantime, but please feel free to comment if any corrections need to be made to my assumptions!
